tests/validation/src/jacobian_validation.cpp

```diff
@@ -0,0 +1,1475 @@
+    const bound_track_parameters<transform3_type>& ref_param,
+    const typename propagator_t::detector_type& det, const field_t& field,
+    const scalar overstep_tolerance, const scalar on_surface_tolerance,
+    const scalar rk_tolerance, const scalar constraint_step,
+    const std::array<scalar, 5u> hs, const scalar avg_step_size) {
+
+    // Return Jacobian
+    bound_track_parameters<transform3_type>::covariance_type eval_jacobian;
+
+    for (unsigned int i = 0u; i < 5u; i++) {
+
+        scalar delta = hs[i];
+
+        if (i == 0 || i == 1) {
+            delta =
+                math_ns::max(avg_step_size * hs[i], 1e-1f * unit<scalar>::mm);
+        } else if (i == 2 || i == 3) {
+            delta = hs[i];
+        } else if (i == 4) {
+            delta = delta * std::abs(ref_param.qop());
+        }
+
+        const auto new_vec0 = get_displaced_bound_vector<propagator_t, field_t>(
+            ref_param, det, field, overstep_tolerance, on_surface_tolerance,
+            rk_tolerance, constraint_step, i, 2.f * delta);
+        const auto new_vec1 = get_displaced_bound_vector<propagator_t, field_t>(
+            ref_param, det, field, overstep_tolerance, on_surface_tolerance,
+            rk_tolerance, constraint_step, i, 1.f * delta);
+        const auto new_vec2 = get_displaced_bound_vector<propagator_t, field_t>(
+            ref_param, det, field, overstep_tolerance, on_surface_tolerance,
+            rk_tolerance, constraint_step, i, -1.f * delta);
+        const auto new_vec3 = get_displaced_bound_vector<propagator_t, field_t>(
+            ref_param, det, field, overstep_tolerance, on_surface_tolerance,
+            rk_tolerance, constraint_step, i, -2.f * delta);
+
+        for (unsigned int j = 0; j < 5u; j++) {
+
+            const scalar v0 = getter::element(new_vec0, j, 0u);
+            const scalar v1 = getter::element(new_vec1, j, 0u);
+            const scalar v2 = getter::element(new_vec2, j, 0u);
+            const scalar v3 = getter::element(new_vec3, j, 0u);
+
+            getter::element(eval_jacobian, j, i) =
+                (-v0 + 8.f * v1 - 8.f * v2 + v3) / (12.f * delta);
+        }
+    }
+
+    return eval_jacobian;
+}
+
+template <typename helix_intersector_t, typename detector_t,
+          typename detector_t::metadata::mask_ids mask_id>
+bound_track_parameters<transform3_type> get_initial_parameter(
+    detector_t& det, const free_track_parameters<transform3_type>& vertex,
+    const vector3& field, const scalar helix_tolerance) {
+
+    // Helix from the vertex
+    detail::helix<transform3_type> hlx(vertex, &field);
+
+    const auto& departure_sf = det.surface(0u);
+    const auto& trf_link = departure_sf.transform();
+    const auto& departure_trf = det.transform_store()[trf_link];
+    const auto& mask_link = departure_sf.mask();
+    const auto& departure_mask =
+        det.mask_store().template get<mask_id>().at(mask_link.index());
+
+    helix_intersector_t hlx_is{};
```

Comment:
Does this work?
```
using departure_mask_t = typename detector_t::masks::template get_type<mask_id>;
using intersection_t = intersection2D<typename detector_t::surface_type,
                           transform3_type>;
helix_intersector<intersection_t, departure_mask_t>hlx_is{};
```
And also in the other places its used

Reply:
Why do you think it is not going to work

Reply:
I meant to ask whether the following would work, sorry

Reply:
It's not working. it says it doesn't have convergence_tolerance

Reply:
That's odd, it should. Oh well, I am going to fix it later then